google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpanTest.java

```diff
@@ -381,7 +426,13 @@ public void transactionRunnerWithError() {
     assertEquals(ErrorCode.INVALID_ARGUMENT, e.getErrorCode());
 
     Map<String, Boolean> spans = failOnOverkillTraceComponent.getSpans();
-    assertThat(spans.size()).isEqualTo(3);
+
+    if (useMultiplexed) {
+      assertThat(spans.size()).isEqualTo(4);
```

Comment:
I'm noticing a flaky behavior due to the new assertions for multiplexed sessions. This is because multiplexed session creation does not block on application startup. Probably we can plug some logic in existing `setWaitForMinSessions` method to also wait for multiplexed session creation? 

To start with this could be used for tests. @olavloite WDYT? This also goes back to our previous discussion that it's ok to block application startup if its an option that customers opt-into and is not switched on by default.